File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/productList/ProductListActivity.java
package com.example.nightwingky.monstone.main_program.productList;

import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.SearchView;
import android.widget.Toast;

import com.example.nightwingky.monstone.R;
import com.example.nightwingky.monstone.main_program.fragmentHome.search.GetJsonData;
import com.example.nightwingky.monstone.main_program.fragmentHome.search.SearchHttp;
import com.example.nightwingky.monstone.main_program.product_Item.ProductActivity;

import org.json.JSONException;

import java.io.IOException;

public class ProductListActivity extends AppCompatActivity implements MyItemClickListener{

    private SearchView mSearchView;
    private RecyclerView rv_list;

    private ProductListAdapter adapter;

    private String queryText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product_list);

        mSearchView = (SearchView) findViewById(R.id.search_product);
        rv_list = (RecyclerView) findViewById(R.id.recycler_product);

        rv_list.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));

        adapter = new ProductListAdapter(this);
        adapter.addList(SearchHttp.mList);
        rv_list.setAdapter(adapter);
        adapter.setOnItemCLickListener(this);
        adapter.notifyDataSetChanged();

        mSearchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                queryText = String.valueOf(mSearchView.getQuery());
                new SearchAsync().execute();
                return true;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                return false;
            }
        });
    }

    @Override
    public void onItemClick(View view, int position) {
        String no = SearchHttp.mList.get(position).getProductId();

        Intent intent = new Intent(ProductListActivity.this, ProductActivity.class);
        intent.putExtra("no", no);
        intent.putExtra("position", position);

        startActivity(intent);
    }

    class SearchAsync extends AsyncTask<String, Void, Boolean> {

        @Override
        protected Boolean doInBackground(String... params) {
            try {
                SearchHttp.mList = GetJsonData.getJsonData(queryText);
                return true;
            } catch (IOException e) {
                e.printStackTrace();
            } catch (JSONException e) {
                e.printStackTrace();
            }

            return false;
        }

        @Override
        protected void onPostExecute(Boolean aBoolean) {
            super.onPostExecute(aBoolean);

            if (!aBoolean) {
                Toast.makeText(ProductListActivity.this, "查找失败", Toast.LENGTH_SHORT).show();
            } else {
                adapter.addList(SearchHttp.mList);
                adapter.notifyDataSetChanged();
                InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                imm.hideSoftInputFromWindow(mSearchView.getWindowToken(), 0) ;
            }
        }
    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/productList/ProductListActivity.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.productList

import android.content.Context
import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.SearchView
import android.widget.Toast


import com.graduation.nightwingky.book_drifting.R
import com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentHome.search.GetJsonData
import com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentHome.search.SearchHttp
import com.graduation.nightwingky.book_drifting.monstone.main_program.productItem.ProductActivity

import org.json.JSONException

import java.io.IOException

class ProductListActivity : AppCompatActivity() {

    private var mSearchView: SearchView? = null
    private var rv_list: RecyclerView? = null

    private var adapter: ProductListAdapter? = null

    private var queryText: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product_list)

        mSearchView = findViewById<View>(R.id.search_product) as SearchView
        rv_list = findViewById<View>(R.id.recycler_product) as RecyclerView

        rv_list!!.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)

        adapter = ProductListAdapter(this)
        adapter!!.addList(SearchHttp.mList)
        rv_list!!.adapter = adapter
        adapter!!.setOnItemClickListener(object : OnRecyclerViewItemListener {
            override fun onItemClick(view: View, data: String) {
                val intent = Intent(this@ProductListActivity, ProductActivity::class.java)
                intent.putExtra("position", data)
                startActivity(intent)
            }
        })
        adapter!!.notifyDataSetChanged()

        mSearchView!!.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                queryText = mSearchView!!.query.toString()
                SearchAsync().execute()
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                return false
            }
        })
    }

    internal inner class SearchAsync : AsyncTask<String, Void, Boolean>() {

        override fun doInBackground(vararg params: String): Boolean? {
            try {
                SearchHttp.mList = GetJsonData.getJsonData(queryText!!)
                return true
            } catch (e: IOException) {
                e.printStackTrace()
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return false
        }

        override fun onPostExecute(aBoolean: Boolean?) {
            super.onPostExecute(aBoolean)

            if ((!aBoolean!!)!!) {
                Toast.makeText(this@ProductListActivity, "查找失败", Toast.LENGTH_SHORT).show()
            } else {
                adapter!!.addList(SearchHttp.mList)
                adapter!!.notifyDataSetChanged()
                val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                imm.hideSoftInputFromWindow(mSearchView!!.windowToken, 0)
            }
        }
    }
}
